internal/api/users/list_test.go

```diff
@@ -0,0 +1,72 @@
+package users
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/model"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+func TestList(test *testing.T) {
+	usersMap := db.MustGetServerUsers()
+
+	users := make([]*model.ServerUser, 0, len(usersMap))
+	for _, user := range usersMap {
+		users = append(users, user)
+	}
+
+	testCases := []struct {
+		email     string
+		pass      string
+		permError bool
+		users     []*model.ServerUser
+	}{
+		// Invalid permissions.
+		{"server-user@test.com", "server-user", true, nil},
+		{"server-creator@test.com", "server-creator", true, nil},
+
+		// Valid permissions.
+		{"server-admin@test.com", "server-admin", false, users},
+		{"admin@test.com", "admin", false, users},
```

Comment:
I think you can leave out `admin@test.com`.
Since they are mainly used in a course context, people may not know right away what their server permissions are.
Instead, how about you just use all the server-* users (which I think would just mean adding `server-owner`).